### PR TITLE
Properly raise FileNotFound even if the dataset is private

### PR DESCRIPTION
**Pull request by lhoestq**
`tests/test_load.py::test_load_streaming_private_dataset` was failing because the hub now returns 401 when getting the HfApi.dataset_info of a dataset without authentication. `load_dataset` was raising ConnectionError, while it should be FileNoteFoundError since it first checks for local files before checking the Hub.

Moreover when use_auth_token is not set (default is False), we should not pass `token=None` to HfApi.dataset_info, or it will use the local token by default - instead it should use no token. It's currently not possible to ask for no token to be used, so as a workaround I simply set token="no-token"

### CODE REVIEW

File: src/datasets/load.py [hunk 1]
```diff
@@ -783,7 +783,7 @@ def get_module(self) -> DatasetModule:
         hfh_dataset_info = HfApi(config.HF_ENDPOINT).dataset_info(
             self.name,
             revision=self.revision,
-            token=token,
+            token=token if token else "no-token",
```
Comment (julien-c) [inline src/datasets/load.py:786]: would passing `token=False` work instead?

Comment (julien-c) [reply src/datasets/load.py:786]: also don't hesitate to ping @SBrandeis / @coyotte508 / @Pierrci on those kind of PRs =)

Comment (SBrandeis) [reply src/datasets/load.py:786]: `"no-token"` won't trigger the type checker so I think it's better

Comment (albertvillanova) [reply src/datasets/load.py:786]: If HFH maintainers prefer `"no-token"`, then this is OK! ;)

File: src/datasets/load.py [hunk 2]
```diff
@@ -1159,7 +1159,10 @@ def dataset_module_factory(
                     elif "401" in str(e):
                         msg = f"Dataset '{path}' doesn't exist on the Hub"
                         msg = msg + f" at revision '{revision}'" if revision else msg
-                        raise FileNotFoundError(msg + ". If the repo is private, make sure you are authenticated.")
+                        raise FileNotFoundError(
+                            msg
+                            + ". If the repo is private, make sure you are authenticated with ``use_auth_token=True`` after logging in with ``huggingface-cli login```."
```
Comment (albertvillanova) [inline src/datasets/load.py:1164]: Two nits:
- I would not use double backticks here: this is written to the shell (no Markdown/reStructuredText rendering is going here)
- Anyway, there is a wrong triple backtick at the end.

**Review by albertvillanova — APPROVED** [review]
Thank you. Good catch! ;)

**Pull request merged**